internal/plugin/server.go

```diff
@@ -331,53 +331,54 @@ func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*plu
 	if err != nil {
 		return nil, fmt.Errorf("failed to get allocate response for CDI: %v", err)
 	}
+	plugin.updateContainerAllocatResponseForMPS(&response, deviceIDs)
```

Comment:
What about when there is no MPS? I haven't read further yet, but the name of this function implies to me that MPS is always in use.

Reply:
There is a check to see whether it is enabled and return as a no-op if not. I elected to pull this into the function, since we may want to extend the logic to cover more compex cases.

Any suggestions on a function name?